pagopa/io-app
# feat: [IOCOM-833] Pull to refresh, new messages' home

⚠️ This PR depends on #5854 ⚠️

## Short description
This PR implements the Pull-to-refresh behaviour on the new messages’home with the new DS.

![Simulator Screenshot - iPhone 15 - 2024-06-16 at 13 20 39](https://github.com/pagopa/io-app/assets/5150343/d8d94d3e-a07c-4181-89c7-6ab22393f9d8)

## List of changes proposed in this pull request
- `CustomRefreshControl` component, given to the `FlatList`, calling the proper method for visibility and the refresh callback
- `getReloadAllMessagesActionForRefreshIfAllowed` in `homeUtils` that implements the business logic to check if a refresh is allowed
- `shouldShowRefreshControllOnListSelector` selector to compute the refresh visibility.
- `isStrictSomeError`, `isSomeLoadingOrSomeUpdating`, `isSomeOrSomeError`, `isLoadingOrUpdating` in `pot` as utility functions
- `getLoadNextPageMessagesActionIfAllowed` in `homeUtils` has been refactored to avoid conflicting parallel requests resulting in inconsistent redux state

## How to test
Using the io-dev-api-server, add some delay to the `GET /messages` endpoint and generate some messages. Use the pull-to-refresh gesture on the list (both INBOX and ARCHIVE) to check that it is working properly (you can start and stop the dev-server without restarting the application to see the messages change)
